core/src/test/scala/integration/kafka/api/SaslSetup.scala

```diff
@@ -43,32 +43,33 @@ trait SaslSetup {
   private val workDir = TestUtils.tempDir()
   private val kdcConf = MiniKdc.createConfig
   private var kdc: MiniKdc = null
+  var serverKeytabFile: Option[File] = None
+  var clientKeytabFile: Option[File] = None
```

Comment:
Introducing this makes the `setJaasConfiguration` method cleaner, but it has two disadvantages:

1. Each var can be mutated from a subclass even though it seems like we never want to do that. We can fix that by making the `var` private and exposing it via a `def`.
2. Each var could be accessed before it's initialised and it may not be obvious. We can improve that by making them `null` to start with so that if someone accesses them before they're initialised, they'll get a NPE. We could do even better by adding some code to the accessor, but it doesn't seem worth it.